pyaedt/modules/PostProcessor.py

```diff
@@ -1122,6 +1122,52 @@ def available_report_quantities(
             )
         return []  # pragma: no cover
 
+    @pyaedt_function_handler()
+    def get_all_report_quantities(
+        self,
+        solutions=None,
+        context=None,
+        is_siwave_dc=False,
+    ):
+        """Return all the possible report categories organized by report types, solution and categories.
+
+        Parameters
+        ----------
+        solutions : str, list optional
+            Solution (or solutions) to get the report quantities.
+            The default is ``None``, in which case the all solutions are used.
+        context : str, dict, optional
+            Report Context.
+            The default is ``None``, in which case the default context is used.
+            For Maxwell 2D/3D Eddy Current solution types this can be provided as a dictionary
+            where the key is the matrix name and value the reduced matrix.
+        is_siwave_dc : bool, optional
+            Whether if the setup is Siwave DCIR or not. Default is ``False``.
+
+        Returns
+        -------
+        dict
+            A dictionary with primary key the report type, secondary key the solution type and
+            third key the report categories.
+        """
+        rep_quantities = {}
+        for rep in self.available_report_types:
+            rep_quantities[rep] = {}
+            if isinstance(solutions, str):
+                solutions = [solutions]
+            elif solutions is None:
+                solutions = self.available_report_solutions(rep)
+            for sol in solutions:
+                rep_quantities[rep][sol] = {}
+                for quant in self.available_quantities_categories(
+                    rep, context=context, solution=sol, is_siwave_dc=is_siwave_dc
```

Comment:
```suggestion
        solutions=None,
        context=None,
        is_siwave_dc=False,
    ):
        """Return all the possible report categories organized by report types, solution and categories.

        Parameters
        ----------
        solutions : list, optional
            Solutions to get the report quantities.
            The default is ``None``, in which case the all solutions are used.
        context : str, dict, optional
            Report Context.
            The default is ``None``, in which case the default context is used.
            For Maxwell 2D/3D Eddy Current solution types this can be provided as a dictionary
            where the key is the matrix name and value the reduced matrix.
        is_siwave_dc : bool, optional
            Whether if the setup is Siwave DCIR or not. Default is ``False``.

        Returns
        -------
        dict
            A dictionary with primary key the report type, secondary key the solution type and
            third key the report categories.
        """
        rep_quantities = {}
        for rep in self.available_report_types:
            rep_quantities[rep] = {}
            if solutions is None:
                solutions = self.available_report_solutions(rep)
            for solution in solutions:
                rep_quantities[rep][solution] = {}
                for quant in self.available_quantities_categories(
                    rep, context=context, solution=solution, is_siwave_dc=is_siwave_dc
```

Reply:
It doesn't make much sense. Most of the time (>90% of the cases) user will enter a single setup or nothing. The way I've implemented still the user can enter more than 1 setup but I would expect not much cases on that.

Reply:
Furthermore the way I've implemented keeps the compatibility with other methods in post (available_quantities_categories, available_report_quantites...) which expects a solution as an input